engine/order_manager.go

```diff
@@ -601,22 +623,72 @@ func (m *OrderManager) processOrders() {
 
 			for z := range result {
 				ord := &result[z]
-				result := m.orderStore.add(ord)
-				if result != ErrOrdersAlreadyExists {
-					msg := fmt.Sprintf("Order manager: Exchange %s added order ID=%v pair=%v price=%v amount=%v side=%v type=%v.",
-						ord.Exchange, ord.ID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type)
+				if !m.Exists(ord) {
+					err = m.Add(ord)
+					if err != nil {
+						log.Errorf(log.OrderMgr,
+							"Order manager: Exchange %s unable to add order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v: %s",
+							ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type, err)
+						continue
+					}
+					msg := fmt.Sprintf("Order manager: Exchange %s added order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v.",
+						ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type)
 					log.Debugf(log.OrderMgr, "%v", msg)
 					m.orderStore.commsManager.PushEvent(base.Event{
 						Type:    "order",
 						Message: msg,
 					})
+					checkedOrderIDs[ord.InternalOrderID] = true
 					continue
+				} else {
+					err = m.UpdateExistingOrder(ord)
+					if err != nil {
+						log.Errorf(log.OrderMgr,
+							"Order manager: Exchange %s unable to update order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v: %s",
+							ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type, err)
+						continue
+					}
+					ord, _ = m.GetByExchangeAndID(ord.Exchange, ord.ID)
+					checkedOrderIDs[ord.InternalOrderID] = true
+					continue
+				}
+			}
+			if !exchanges[i].GetBase().GetSupportedFeatures().RESTCapabilities.GetOrder {
+				continue
+			}
+			for x := range ordersToCheck {
+				curTime := time.Now()
+				// Only check if at least 1 minute not updated
+				if curTime.Sub(ordersToCheck[x].LastUpdated) < time.Minute {
+					checkedOrderIDs[ordersToCheck[x].InternalOrderID] = true
+				} else if checkedOrderIDs[ordersToCheck[x].InternalOrderID] == false {
+					log.Debugf(log.OrderMgr, "Order manager: Exchange %s Unchecked order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v",
+						ordersToCheck[x].Exchange, ordersToCheck[x].ID, ordersToCheck[x].InternalOrderID, ordersToCheck[x].Pair, ordersToCheck[x].Price,
+						ordersToCheck[x].Amount, ordersToCheck[x].Side, ordersToCheck[x].Type)
+					go m.FetchAndUpdateExchangeOrder(exchanges[i], &ordersToCheck[x], supportedAssets[y], curTime)
 				}
 			}
 		}
 	}
 }
 
+func (m *OrderManager) FetchAndUpdateExchangeOrder(exch exchange.IBotExchange, ord *order.Detail, assetType asset.Item, curTime time.Time) {
+	fetchedOrder, err := exch.GetOrderInfo(ord.ID, ord.Pair, assetType)
+	if err != nil {
+		log.Errorf(log.OrderMgr, "Unable to get additional order info: %s", err.Error())
+		ord.Status = order.UnknownStatus
+		return
+	}
+	fetchedOrder.LastUpdated = curTime
```

Comment:
fetch request above might take some time due to rate limit etc. MIght be better to do time.Now() here.